File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Persistencia_de_Objetos/Ejercicio1/Java/ArchivoEmpleado.java
package Persistencia_de_Objetos.Ejercicio1.Java;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class ArchivoEmpleado {
    private String nomArchivo;

    public ArchivoEmpleado(String nomArchivo) {
        this.nomArchivo = nomArchivo;
        crearArchivo(); 
    }

    private void crearArchivo() {
        File file = new File(nomArchivo);
        if (!file.exists()) {
            try {
                if (file.createNewFile()) {
                    System.out.println("Archivo '" + nomArchivo + "' creado exitosamente");
                } else {
                    System.err.println("No se pudo crear el archivo '" + nomArchivo );
                }
            } catch (IOException e) {
                System.err.println("Error al crear el archivo '" + nomArchivo + "': " + e.getMessage());
            }
        } else {
            System.out.println("Archivo '" + nomArchivo + "' ya existe");
        }
    }

    // a) 
    public void guardarEmpleado(Empleado e) {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(nomArchivo, true))) {
            writer.write(e.toString());
            writer.newLine(); 
            System.out.println("Empleado '" + e.getNombre() + "' guardado exitosamente en '" + nomArchivo );
        } catch (IOException error) {
            System.err.println("Error al guardar empleado en '" + nomArchivo + "': " + error.getMessage());
        }
    }

    private List<Empleado> cargarTodosEmpleados() {
        List<Empleado> empleados = new ArrayList<>();
        try (BufferedReader reader = new BufferedReader(new FileReader(nomArchivo))) {
            String linea;
            while ((linea = reader.readLine()) != null) {
                Empleado empleado = Empleado.fromString(linea);
                if (empleado != null) {
                    empleados.add(empleado);
                }
            }
        } catch (IOException e) {
            System.err.println("Error al leer el archivo '" + nomArchivo + "': " + e.getMessage());
        }
        return empleados;
    }

    // b) 
    public Empleado buscaEmpleado(String nombreBuscar) {
        List<Empleado> empleados = cargarTodosEmpleados();
        for (Empleado empleado : empleados) {
            if (empleado.getNombre().equalsIgnoreCase(nombreBuscar)) { 
                return empleado;
            }
        }
        return null; 
    }

    // c)
    public Empleado mayorSalario(float sueldoMinimo) {
        List<Empleado> empleados = cargarTodosEmpleados();
        for (Empleado empleado : empleados) {
            if (empleado.getSalario() > sueldoMinimo) {
                return empleado;
            }
        }
        return null; 
    }

    public void limpiarArchivo() {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(nomArchivo))) {
            writer.write(""); 
            System.out.println("Contenido del archivo '" + nomArchivo + "' limpiado.");
        } catch (IOException e) {
            System.err.println("Error al limpiar el archivo '" + nomArchivo + "': " + e.getMessage());
        }
    }
}

File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Persistencia_de_Objetos/Ejercicio1/Python/ArchivoEmpleado.py
import os
from typing import List, Optional
from Empleado import Empleado

class ArchivoEmpleado:
    
    def __init__(self, nom_archivo: str):
        self.nom_archivo = nom_archivo
        self.crear_archivo() 

    def crear_archivo(self):
        
        if not os.path.exists(self.nom_archivo):
            try:
                with open(self.nom_archivo, 'w') as f:
                    pass 
                print(f"Archivo '{self.nom_archivo}' creado exitosamente.")
            except IOError as e:
                print(f"Error al crear el archivo '{self.nom_archivo}': {e}")
        else:
            print(f"Archivo '{self.nom_archivo}' ya existe.")

    # a) 
    def guardar_empleado(self, e: Empleado):
        
        try:
            with open(self.nom_archivo, 'a') as f: 
                f.write(str(e) + '\n')
            print(f"Empleado '{e.nombre}' guardado exitosamente en '{self.nom_archivo}'.")
        except IOError as error:
            print(f"Error al guardar empleado en '{self.nom_archivo}': {error}")

    def _cargar_todos_empleados(self) -> List[Empleado]:
        
        empleados = []
        try:
            with open(self.nom_archivo, 'r') as f:
                for linea in f:
                    empleado = Empleado.from_string(linea)
                    if empleado:
                        empleados.append(empleado)
            return empleados
        except FileNotFoundError:
            print(f"El archivo '{self.nom_archivo}' no se encontro")
            return []
        except IOError as e:
            print(f"Error al leer el archivo '{self.nom_archivo}': {e}")
            return []

    # b) 
    def busca_empleado(self, nombre_buscar: str) -> Optional[Empleado]:
       
        empleados = self._cargar_todos_empleados()
        for empleado in empleados:
            if empleado.nombre.lower() == nombre_buscar.lower():
                return empleado
        return None

    # c) 
    def mayor_salario(self, sueldo_minimo: float) -> Optional[Empleado]:
       
        empleados = self._cargar_todos_empleados()
        for empleado in empleados:
            if empleado.salario > sueldo_minimo:
                return empleado
        return None

    def limpiar_archivo(self):
        
        try:
            with open(self.nom_archivo, 'w') as f:
                f.write('')
            print(f"Contenido del archivo '{self.nom_archivo}' limpiado.")
        except IOError as e:
            print(f"Error al limpiar el archivo '{self.nom_archivo}': {e}")